ward/testing.py

```diff
@@ -104,12 +105,56 @@ class Test:
     sout: StringIO = field(default_factory=StringIO)
     serr: StringIO = field(default_factory=StringIO)
     ward_meta: WardMeta = field(default_factory=WardMeta)
+    record_duration: bool = True
+    timer: Optional["Timer"] = None
+    result: Optional["TestResult"] = None
```

Comment:
Do we have to store the `result` in the `Test`? Could we just return the `TestResult` from `Test::run`?

Reply:
I think that's fine. I was thinking that the duration is more a property of a Test than of a TestResult.

Reply:
I wonder if we could solidify the abstraction a bit going forward and say:

* `Test` contains any data known before the test runs.
* `TestResult` contains any data that is only calculated during the run of the test.

Maybe that, or something similar to it, could be a good guideline going forward?

Reply:
Yeah, this seems reasonable